### PR TITLE
EDX-376: MIN | UNCLAIM displays where CLAIM is expected when ministry…

### PR DESCRIPTION
**Pull request by TimOsse**
… user updates secure message claim

### CODE REVIEW

File: frontend/src/components/secure-message/MessageDisplay.vue [hunk 1]
```diff
@@ -102,9 +102,10 @@
...
                   <span class="ml-1 markAsSpan">{{`Mark As ${secureExchange.isReadByMinistry ? 'Unread' : 'Read'}` }}</span>
                 </v-btn>
-                <v-btn id="claimAsButton" class="my-4 mx-2" v-on:click="clickClaimMsgButton" :disabled="!isEditable()">
-                  <v-icon>{{ secureExchange.reviewer ? 'mdi-account-off-outline' : 'mdi-account-check-outline' }}</v-icon>
-                  <span class="ml-1">{{ secureExchange.reviewer ? 'Unclaim' : 'Claim' }}</span>
+                <v-btn id="claimAsButton" class="my-4 mx-2" v-on:click="clickClaimMsgButton" :disabled="!isEditable() && !isClaimable() && !isUnClaimable()">
+                  <v-icon>{{ !isClaimable() ? 'mdi-account-off-outline' : 'mdi-account-check-outline' }}</v-icon>
+                  <span v-if="isClaimable()" class="ml-1">Claim</span>
+                  <span v-if="!isClaimable()" class="ml-1">Unclaim</span>
                 </v-btn>
```
Comment (SoLetsDev) [inline frontend/src/components/secure-message/MessageDisplay.vue:109]: I think we can combine the spans on lines 107 and 108
 <span class="ml-1">{{ isClaimable() ? 'Claim' : 'Unclaim' }}</span>
 
 Unless there's some issue with inline conditional statements for vue text.

Comment (TimOsse) [reply frontend/src/components/secure-message/MessageDisplay.vue:109]: Definitely, I don't know why i split them up.

File: frontend/src/components/secure-message/MessageDisplay.vue [hunk 2]
```diff
@@ -507,16 +508,35 @@ export default {
...
     },
+    isClaimable(){
+      if(this.secureExchange.reviewer === ''){
+        return true;
+      }
+      if(this.secureExchange.reviewer !== '' && this.secureExchange.reviewer !== this.userInfo.userName){
+        return true;
+      } else {
+        return false;
+      }
```
Comment (SoLetsDev) [inline frontend/src/components/secure-message/MessageDisplay.vue:519]: Is there a way to simplify the flow for this? 
I'm thinking like a stepwise check 
Line 512 we check whether or not a reviewer is there. If not the message is always claimable. 

Line 515 If the reviewer is not the same as user. Then return true 

Then in all other cases it's false 

I'm thinking something like this 
 isClaimable(){
      if(this.secureExchange.reviewer === ''){
        return true;
      }
      if(this.secureExchange.reviewer !== this.userInfo.userName){
        return true;
      } 
        return false;
    

^^^ removes if else statement

Comment (arcshiftsolutions) [reply frontend/src/components/secure-message/MessageDisplay.vue:519]: @SoLetsDev You can simplify further:

`isClaimable(){
  return this.secureExchange.reviewer === '';
}`

Comment (TimOsse) [reply frontend/src/components/secure-message/MessageDisplay.vue:519]: > @SoLetsDev You can simplify further:
> 
> `isClaimable(){ return this.secureExchange.reviewer === ''; }`

But the user needs to be able to claim it even when someone else has already. According to the ticket.

Comment (arcshiftsolutions) [reply frontend/src/components/secure-message/MessageDisplay.vue:519]: @TimOsse - No worries, we can just make it a double condition like this:

`
isClaimable(){ return this.secureExchange.reviewer === '' && this.secureExchange.reviewer !== this.userInfo.userName; }
`

**Review by SoLetsDev — APPROVED** [review]

**Pull request merged**